expectRetrying.ts

```diff
@@ -406,6 +412,67 @@ export function createExpectation(
       );
     },
 
+    async toHaveAttribute(attribute: string, expectedValue?: string) {
+      const matcherName = "toHaveAttribute";
+
+      const stacktrace = parseStackTrace(new Error().stack);
+      const executionContext = captureExecutionContext(stacktrace);
+
+      if (!executionContext) {
+        throw new Error("k6 failed to capture execution context");
+      }
+
+      const renderer = MatcherErrorRendererRegistry.getRenderer(matcherName);
+      const renderConfig = MatcherErrorRendererRegistry.getConfig();
+
+      try {
+        await withRetry(async () => {
+          const result = await toHaveAttribute(
+            locator,
+            attribute,
+            expectedValue,
+          );
+          const finalResult = isNegated ? result.negate() : result;
+
+          if (finalResult.passed) {
+            return;
+          }
+
+          const info: MatcherErrorInfo = {
+            executionContext,
+            matcherName,
+            expected: finalResult.detail.expected,
+            received: finalResult.detail.received,
+          };
+
+          usedAssert(
+            false,
+            renderer.render(info, renderConfig),
+            isSoft,
+            config.softMode,
+          );
+        }, retryConfig);
+      } catch {
+        const info: MatcherErrorInfo = {
+          executionContext,
+          matcherName,
+          expected: "An element matching the locator.",
+          received:
+            `Timeout waiting for element matching locator (${retryConfig.timeout}ms)`,
+        };
+
+        usedAssert(
+          false,
+          MatcherErrorRendererRegistry.getRenderer(matcherName).render(
+            info,
+            MatcherErrorRendererRegistry.getConfig(),
+          ),
+          isSoft,
+          config.softMode,
+        );
+      }
```

Comment:
The only things that would differ between implementations is the `matcherName` and which function is being called, so this could easily be modified to be reusable.